solidity/contracts/hooks/StaticProtocolFee.sol

```diff
@@ -109,6 +84,30 @@ contract StaticProtocolFee is IPostDispatchHook, Ownable {
 
     // ============ Internal Functions ============
 
+    /// @inheritdoc AbstractPostDispatchHook
+    function _postDispatch(bytes calldata, bytes calldata message)
+        internal
+        override
+    {
+        require(
+            msg.value >= protocolFee,
+            "StaticProtocolFee: insufficient protocol fee"
+        );
+
+        uint256 refund = msg.value - protocolFee;
+        if (refund > 0) payable(message.senderAddress()).sendValue(refund);
```

Comment:
should we use metadata refund address here and fallback to message.senderAddress?